docker-compose.yml

```diff
@@ -1,4 +1,4 @@
-version: "3"
+version: "2.4"
```

Comment:
why downgrade?

Reply:
I switched the the latest v2 release, so that I could make the ratelimit service depend on the redis service being healthy.

docker-compose removed support for depends_on conditions in v3 because it does not work with swarm mode.

Their current recommendation is to use v3 and have the application container await the upstream service dependency, which is more troublesome to implement.

Since we do not need swarm mode, I thought using v2 was the easier way to get the desired behavior.

Reply:
Here's a reference: https://github.com/docker/compose/issues/4305#issuecomment-305378202